wgengine/magicsock/magicsock.go

```diff
@@ -231,6 +231,7 @@ type Conn struct {
 	// This block mirrors the contents and field order of the Options
 	// struct. Initialized once at construction, then constant.
 
+	interEp                netip.AddrPort
```

Comment:
what's the locking rules for reading/writing this? It shouldn't be in this section, which is static stuff.

Is it guarded by mu? Then it needs to go below mu (after line 320).

But I don't think that's currently the case. It seems the current reading/writing is a data race.

Also, this field needs docs on what it is. It isn't obvious from the name.

Reply:
> what's the locking rules for reading/writing this? It shouldn't be in this section, which is static stuff.
> 
> Is it guarded by mu? Then it needs to go below mu (after line 320).
> 
> But I don't think that's currently the case. It seems the current reading/writing is a data race.
> 
> Also, this field needs docs on what it is. It isn't obvious from the name.

Not sure I'm able to answer your question, not quit familiar with the general locking shceme here. 
The intention of this field is to temporarly store this interim endpoint information received from Ping message and use it when it's time to update local endpoint list.